frontend/src/components/common/Resource/PortForward.tsx

```diff
@@ -0,0 +1,174 @@
+    const serviceName = !isPod ? massagedResourceName : '';
+    const podName = isPod ? massagedResourceName : '';
+    startPortForward(cluster, namespace, podName, containerPort, serviceName)
+      .then((data: any) => {
+        setPortForward(data);
+        setIsCurrentPodOrServicePortForwarding(true);
+        const portForwards = localStorage.getItem(
+          isPod ? PORTFORWARD_STORAGE_KEY_FOR_PODS : PORTFORWARD_STORAGE_KEY_FOR_SERVICES
+        );
+        if (!portForwards) {
+          const portForwardsList = [];
+          portForwardsList.push(JSON.stringify(data));
+          return;
+        }
+        const parsedPortForwards = JSON.parse(portForwards);
+        parsedPortForwards.push(data);
+        localStorage.setItem(
+          isPod ? PORTFORWARD_STORAGE_KEY_FOR_PODS : PORTFORWARD_STORAGE_KEY_FOR_SERVICES,
+          JSON.stringify(parsedPortForwards)
+        );
+      })
+      .catch(error => {
+        setError(error);
+        setPortForward(null);
+      });
+  }
+
+  function portForwardStopHandler() {
+    if (!portForward || !cluster) {
+      return;
+    }
+    deletePortForward(
+      cluster,
+      portForward.id,
+      isPod ? PODS_BACKEND_PORTFORWARD_KEY : SERVICE_BACKEND_PORTFORWARD_KEY
+    )
+      .then(() => {
+        setPortForward(null);
+        const id = portForward.id;
+        const portForwards = localStorage.getItem(
+          isPod ? PORTFORWARD_STORAGE_KEY_FOR_PODS : PORTFORWARD_STORAGE_KEY_FOR_SERVICES
+        );
+        if (!portForwards) {
+          return;
+        }
+        const parsedPortForwards = JSON.parse(portForwards);
+        const filteredPortForwards = parsedPortForwards.filter((item: any) => item.id !== id);
+
+        localStorage.setItem(
+          isPod ? PORTFORWARD_STORAGE_KEY_FOR_PODS : PORTFORWARD_STORAGE_KEY_FOR_SERVICES,
+          JSON.stringify(filteredPortForwards)
+        );
+      })
+      .catch(error => {
+        setError(error);
+        setPortForward(null);
+      });
+  }
+
+  React.useEffect(() => {
+    if (!cluster) {
+      return;
+    }
+    listPortForward(
+      cluster,
+      isPod ? PODS_BACKEND_PORTFORWARD_KEY : SERVICE_BACKEND_PORTFORWARD_KEY
+    ).then(result => {
+      const portforwards = result;
+      if (!portforwards || Object.entries(portforwards).length === 0) {
+        return;
+      }
+      for (const key in portforwards) {
+        const item = portforwards[key];
+        if (
+          item.namespace === namespace &&
+          (item.pod === name || item.service === name) &&
+          item.cluster === cluster
+        ) {
+          setIsCurrentPodOrServicePortForwarding(true);
+          setPortForward(portforwards[key]);
+          return;
+        }
+      }
+    });
+  }, []);
+
+  if (isPod && !isPodRunning) {
+    return null;
+  }
+
+  return !portForward ? (
+    <Box display="flex">
+      <Button onClick={handlePortForward}>{t('frequent|Port Forward')}</Button>
```

Comment:
This needs to be allow only one click as I could do several clicks before the button disappears which allowed me to create 4 working port forwardings, only one of which (the last one) was visible in the UI.

Reply:
updated now it shows a loader when we start the portforward